ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -139,6 +154,32 @@ public void run() {
   }
 
   /**
+   * Pauses hard deletes. If hard deletes is half way in hard deleting a range, it will continue to finish hard deleting
+   * the range and then pause. Hard delete will be in paused state until {@link #resume()} is called.
+   */
+  void pause() throws InterruptedException {
+    if (!paused.getAndSet(true)) {
+      synchronized (lock) {
+        logger.info("HardDelete thread has been paused ");
+        // notify if hard delete thread is waiting after caught up
+        lock.notify();
```

Comment:
why does this need to be done? If the thread wakes up in its own time, it will immediately see the pause variable and pause anyway right?